### PR TITLE
add mfpca.face

### PR DESCRIPTION
**Pull request by ecui1**
We added the mfpca.face() function to refund, which implements the fast MFPCA method that we have been working on. We checked the package and it worked well on our laptops and passed R CMD check.

### CODE REVIEW

**Comment by fabian-s** [comment]
Thank you for the pull request, that's great.
 
Please add some unit tests for the new code that uses its basic functionality on some simple examples. 

This also gives us a new dependency on `simex`,  for using just a single function. Would it be possible to avoid `simex::diag.block` and use a straight kronecker product or `Matrix::bdiag`? We already depend on `Matrix` anyway, so either of those would be preferable.

File: R/mfpca.face.R [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+  
+  pspline.setting.mfpca <- function(x,knots=35,p=3,m=2,weight=NULL,type="full",
+                                    knots.option="equally-spaced"){
+    # x: the marginal data points
+    # knots: the list of interior knots or the numbers of interior knots
+    # p: degrees for B-splines, with defaults values 3
+    # m: orders of difference penalty, with default values 2
+    # knots.option: type of knots placement, with default values "equally-spaced"
+    
+    #require(splines)
```
Comment (fabian-s) [inline R/mfpca.face.R]: please remove unnecessary, commented out code

File: R/mfpca.face.R [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+    
+    if(is.null(weight)) weight <- rep(1,length(x))
+    
+    if(type=="full"){
+      
+      Sig = crossprod(matrix.multiply.mfpca(B,weight,option=2),B)
+      eSig = eigen(Sig)
+      V = eSig$vectors
+      E = eSig$values
+      if(min(E)<=0.0000001) {#cat("Warning! t(B)%*%B is singular!\n");
```
Comment (fabian-s) [inline R/mfpca.face.R]: see above -- plz remove or re-activate these warnings (and use `warning` or `message`, not `cat`)

**Comment by ecui1** [comment]
Thanks for the review. We have updated the function and pushed it to the forked repo.

We have added some unit tests to 'test-mfpca.R' in the tests folder, and removed the dependency on simex package. 

The new function works well, and the package passed R CMD check.

Please let us know if you have any other comments/questions.

**Review by fabian-s — APPROVED** [review]
Thanks, that was quick!

**Comment by fabian-s** [comment]
I'll let @julia-wrobel  take care of the rest ... ;)

File: R/mfpca.face.R (hunk 1 above)
Comment (ecui1) [reply R/mfpca.face.R]: The commented code is removed now.

File: R/mfpca.face.R (hunk 2 above)
Comment (ecui1) [reply R/mfpca.face.R]: These warning messages are removed now from the code.

**Comment by julia-wrobel** [comment]
Thanks for reviewing @fabian-s!

**Pull request merged**